tools/generate-helpers/pg-table-bindings/store.go.tpl

```diff
@@ -893,12 +897,34 @@ func (s *storeImpl) DeleteMany(ctx context.Context, ids []{{$singlePK.Type}}) er
     {{- end }}
     {{- end }}{{/* if not $inMigration */}}
 
-    q := search.ConjunctionQuery(
-    sacQueryFilter,
-        search.NewQueryBuilder().AddDocIDs(ids...).ProtoQuery(),
-    )
+    // Batch the deletes
+    batchSize := deleteBatchSize
+    numRecordsToDelete := len(ids)
+    for {
+        if len(ids) == 0 {
+            break
+        }
+
+        if len(ids) < batchSize {
+            batchSize = len(ids)
+        }
+
+        idBatch := ids[:batchSize]
+        q := search.ConjunctionQuery(
+        sacQueryFilter,
+            search.NewQueryBuilder().AddDocIDs(idBatch...).ProtoQuery(),
+        )
+
+        if err := postgres.RunDeleteRequestForSchema(ctx, schema, q, s.db); err != nil {
+            log.Errorf("unable to delete all the records.  Successfully deleted %d out of %d", numRecordsToDelete - len(ids), numRecordsToDelete)
```

Comment:
You can wrap the error first and then log  and return it. So you do not need to have the same message here twice.